### PR TITLE
[Enhancement] Add more explain info for HdfsScanNode

### LINKED ISSUE #10069: Query on hive table lost partition predicates
### Steps to reproduce the behavior (Required)

create hive table with partition column, and create catalog on it

> create table parttbl (c0 int, c1 string) partitioned by (`date` string) stored as orc;

running following query

> explain select * from parttbl where date = '20120202' and c0 > 0;

```
+------------------------------------------+
| Explain String                           |
+------------------------------------------+
| PLAN FRAGMENT 0                          |
|  OUTPUT EXPRS:1: c0 | 2: c1 | 3: date    |
|   PARTITION: UNPARTITIONED               |
|                                          |
|   RESULT SINK                            |
|                                          |
|   1:EXCHANGE                             |
|                                          |
| PLAN FRAGMENT 1                          |
|  OUTPUT EXPRS:                           |
|   PARTITION: RANDOM                      |
|                                          |
|   STREAM DATA SINK                       |
|     EXCHANGE ID: 01                      |
|     UNPARTITIONED                        |
|                                          |
|   0:HdfsScanNode                         |
|      TABLE: parttbl                      |
|      NON-PARTITION PREDICATES: 1: c0 > 0 |
|      MIN/MAX PREDICATES: 4: c0 > 0       |
|      partitions=0/0                      |
|      cardinality=1                       |
|      avgRowSize=3.0                      |
|      numNodes=0                          |
+------------------------------------------+
```

### Expected behavior (Required)

We expect to see partition predicate like `date = 20120202`

### Real behavior (Required)

but there  is no partition predicate.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

```
MySQL [emr_test.zya]> select current_version();
+-------------------+
| current_version() |
+-------------------+
| ZHANGYAN a8bad07  |
+-------------------+
1 row in set (0.04 sec)
```

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10069 

## Problem Summary(Required) ：

Add NO EVAL-PARTITION PREDICATES for explain hdfs scan node

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 16 (81.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/ListPartitionPruner.java|0|1|00.00%|[102]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|4|6|66.67%|[127, 128]|
|:large_blue_circle:|com/starrocks/catalog/PartitionKey.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|5|5|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/HivePartitionPruneTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package com.starrocks.sql.plan;
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/HivePartitionPruneTest.java:1]: Pls add license header

**Review by imay — COMMENTED** [review]
License header

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/HivePartitionPruneTest.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/HivePartitionPruneTest.java:3]: Done

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 16 (81.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/ListPartitionPruner.java|0|1|00.00%|[102]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|4|6|66.67%|[127, 128]|
|:large_blue_circle:|com/starrocks/catalog/PartitionKey.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|5|5|100.00%|[]|

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Review by dirtysalt — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Pull request merged**